sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -215,22 +228,24 @@ def list_jobs(self, endpoint_name: str, *, name: str = None) -> ItemPaged[BatchJ
         :rtype: ItemPaged[BatchJob]
         """
 
-        workspace_operations = self._all_operations.all_operations[AzureMLResourceType.WORKSPACE]
-        mfe_base_uri = _get_mfe_base_url_from_discovery_service(
-            workspace_operations, self._workspace_name, self._requests_pipeline
+
+        self._batch_job_deployment = self._dataset_dataplane_client.batch_job_deployment
+
+
+        return self._batch_job_deployment.list(
+            endpoint_name=endpoint_name,
+            deployment_name=name,
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            cls=lambda objs: [BatchJob._from_rest_object(obj) for obj in objs],
+            **self._init_kwargs,
         )
 
-        with modified_operation_client(self._batch_job_deployment, mfe_base_uri):
-            result = self._batch_job_deployment.list(
-                endpoint_name=endpoint_name,
-                deployment_name=name,
-                resource_group_name=self._resource_group_name,
-                workspace_name=self._workspace_name,
-                **self._init_kwargs,
-            )
 
-            # This is necessary as the paged result need to be resolved inside the context manager
-            return list(result)
+    def _get_workspace_location(self) -> str:
```

Comment:
This is duplicated. Same function as the one declared in line 250. Please remove these changes.